src/components/MarkdownWrapper/MarkdownWrapper.style.tsx

```diff
@@ -51,10 +52,36 @@ export const MarkdownWrapperStyled = styled.div.attrs<MarkdownWrapperStyledProps
   }
 `;
 
-export const ButtonLink = styled.a.attrs((props) => ({
-  ...props,
-  className: `fr-btn ${props.className || ''}`,
-}))``;
+type TagName = keyof typeof markupData;
+type ExtraEventType = {
+  children: React.ReactNode;
+  className?: string;
+  tagName?: TagName;
+  trackEvent?: string;
+};
+
+export const ButtonLink = styled.a.attrs(
+  (props: React.HTMLProps<HTMLLinkElement> & ExtraEventType) => {
+    const { className, tagName, trackEvent } = props;
+    const getMatomoEventKey = (tagName?: TagName) =>
```

Comment:
Why are you creating a function for that ?

Reply:
Just for visibility.